gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/v4/impl/ApiServiceImpl.java

```diff
@@ -562,22 +562,20 @@ public void delete(ExecutionContext executionContext, String apiId, boolean clos
 
             Set<GenericPlanEntity> plans = planSearchService.findByApi(executionContext, apiId);
             if (closePlans) {
-                plans =
-                    plans
-                        .stream()
-                        .filter(plan -> plan.getPlanStatus() != PlanStatus.CLOSED)
-                        .map(plan -> planService.close(executionContext, plan.getId()))
-                        .collect(Collectors.toSet());
-            }
-
-            Set<String> plansNotClosed = plans
-                .stream()
-                .filter(plan -> plan.getPlanStatus() == PlanStatus.PUBLISHED)
-                .map(GenericPlanEntity::getName)
-                .collect(toSet());
-
-            if (!plansNotClosed.isEmpty()) {
-                throw new ApiNotDeletableException(plansNotClosed);
+                plans
+                    .stream()
+                    .filter(plan -> plan.getPlanStatus() != PlanStatus.CLOSED)
+                    .forEach(plan -> planService.close(executionContext, plan.getId()));
```

Comment:
Just wondering what if a plan is not "well closed"... Is it possible? What would be the status? Maybe an exception is raised?
Could you check please?